extensions/wikia/CoppaTool/RenameIPProcess.class.php

```diff
@@ -0,0 +1,1094 @@
+			);
+
+			$conds = array(
+				$extra['username_column'] => $oldusername,
+			);
+
+			if ( !empty( $extra['userid_column'] ) ) {
+				$conds[$extra['userid_column']] = $uid;
+			}
+
+			if ( !empty( $extra['conds'] ) ) {
+				$conds = array_merge( $extra['conds'], $conds );
+			}
+
+			$opts = array(
+				'LIMIT' => self::MAX_ROWS_PER_QUERY,
+			);
+
+			$affectedRows = 1;
+
+			while ( $affectedRows > 0 ) {
+				$dbw->update( $table, $values, $conds, __CLASS__ . '::' . __METHOD__, $opts );
+				$affectedRows = $dbw->affectedRows();
+				$this->addLog( "SQL: " . $dbw->lastQuery() );
+				$dbw->commit();
+				$this->addLog( "In {$dbName}.{$table}.{$extra['username_column']} {$affectedRows} row(s) was(were) updated." );
+				sleep( USERRENAME_LOOP_PAUSE );
+			}
+		} catch ( Exception $e ) {
+			$this->addLog( "Exception in renameInTable(): " . $e->getMessage() . ' in ' . $e->getFile() . ' at line ' . $e->getLine() );
+		}
+
+		$this->addLog( "Finished processing {$dbName}.{$table}.{$extra['username_column']}." );
+
+		return true;
+	}
+
+	/**
+	 * Reset local editcount for renamed user and fake user
+	 * @author Kamil Koterba
+	 * @since Feb 2014
+	 */
+	private function resetEditCountWiki() {
+		// Renamed user
+		$uss = new UserStatsService( $this->mUserId );
+		$uss->calculateEditCountWiki();
+
+		// FakeUser
+		if ( $this->mFakeUserId != 0 ) {
+			$uss = new UserStatsService( $this->mFakeUserId );
+			$uss->calculateEditCountWiki();
+		} else {
+			// use OldUsername if FakeUser isn't set
+			$oldUser = User::newFromName( $this->mOldUsername );
+			$uss = new UserStatsService( $oldUser->getId() );
+			$uss->calculateEditCountWiki();
+		}
+	}
+
+	/**
+	 * @author Federico "Lox" Lucignano <federico@wikia-inc.com>
+	 * Performs action for cleaning up temporary data at the very end of a process
+	 */
+	public function cleanup() {
+		if ( $this->mFakeUserId ) {
+			$this->addLog( "Cleaning up process data in user option renameData for ID {$this->mFakeUserId}" );
+
+			$fakeUser = User::newFromId( $this->mFakeUserId );
+			$fakeUser->setGlobalAttribute( 'renameData', self::RENAME_TAG . '=' . $this->mNewUsername );
+			$fakeUser->saveSettings();
+			$fakeUser->saveToCache();
+		}
+
+		// TODO: Add a hook
+		$hookName = 'UserRename::Cleanup';
+		$this->addLog( "Broadcasting hook: {$hookName}" );
+		Hooks::run( $hookName, array( $this->mRequestorId, $this->mRequestorName, $this->mUserId, $this->mOldUsername, $this->mNewUsername ) );
+
+		$tasks = [];
+		if ( isset( $this->mLogTask ) ) {
+			$tasks[] = $this->mLogTask->getID();
+		}
+
+		$this->addMainLog( "finish", RenameUserLogFormatter::finish( $this->mRequestorName, $this->mOldUsername, $this->mNewUsername, $this->mReason, $tasks ) );
+	}
+
+	/**
+	 * Sends the internal log message to the specified destination
+	 *
+	 * @param $text string Log message
+	 * @param $arg1 mixed Multiple format parameters
+	 */
+	public function addLog( $text, $arg1 = null ) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$arg1". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

[project-id]: wikia:app